admin/admin-filters-sanitize.php

```diff
@@ -0,0 +1,132 @@
+		if ( current_user_can( 'edit_posts' ) && null !== $request->get_param( 'is_block_editor' ) ) {
+			// When it's a post request on a new post the language is sent into the request.
+			if ( ! empty( $request->get_param( 'lang' ) ) ) {
+				$this->curlang = $this->model->get_language( sanitize_key( $request->get_param( 'lang' ) ) );
+			} else {
+				// Otherwise we need to get the language from the post itself.
+				$this->curlang = $this->model->post->get_language( sanitize_key( $request->get_param( 'id' ) ) );
+			}
+		}
+		return $result;
+	}
+
+	/**
+	 * Filters the locale according to the current language instead of the language
+	 * of the admin interface
+	 *
+	 * @since 2.0
+	 *
+	 * @param string $locale
+	 * @return string
+	 */
+	public function get_locale( $locale ) {
+		if ( isset( $_POST['post_lang_choice'] ) && $lang = $this->model->get_language( sanitize_key( $_POST['post_lang_choice'] ) ) ) { // phpcs:ignore WordPress.Security.NonceVerification
+			$locale = $lang->locale;
+		} elseif ( isset( $_POST['term_lang_choice'] ) && $lang = $this->model->get_language( sanitize_key( $_POST['term_lang_choice'] ) ) ) { // phpcs:ignore WordPress.Security.NonceVerification
+			$locale = $lang->locale;
+		} elseif ( isset( $_POST['inline_lang_choice'] ) && $lang = $this->model->get_language( sanitize_key( $_POST['inline_lang_choice'] ) ) ) { // phpcs:ignore WordPress.Security.NonceVerification
+			$locale = $lang->locale;
+		} elseif ( ! empty( $this->curlang ) ) {
+			$locale = $this->curlang->locale;
+		}
+
+		return $locale;
+	}
+
+	/**
+	 * Maybe fix the result of sanitize_title() in case the languages include German or Danish
+	 * Without this filter, if language of the title being sanitized is different from the language
+	 * used for the admin interface and if one this language is German or Danish, some specific
+	 * characters such as ä, ö, ü, ß are incorrectly sanitized.
+	 *
+	 * @since 2.0
+	 *
+	 * @param string $title     Sanitized title.
+	 * @param string $raw_title The title prior to sanitization.
+	 * @param string $context   The context for which the title is being sanitized.
+	 * @return string
+	 */
+	public function sanitize_title( $title, $raw_title, $context ) {
+		static $once = false;
+
+		if ( ! $once && 'save' == $context && ! empty( $title ) ) {
+			$once = true;
+			add_filter( 'locale', array( $this, 'get_locale' ), 20 ); // After the filter for the admin interface
+			$title = sanitize_title( $raw_title, '', $context );
+			remove_filter( 'locale', array( $this, 'get_locale' ), 20 );
+			$once = false;
+		}
+		return $title;
+	}
+
+	/**
+	 * Maybe fix the result of sanitize_user() in case the languages include German or Danish
+	 *
+	 * @since 2.0
+	 *
+	 * @param string $username     Sanitized username.
+	 * @param string $raw_username The username prior to sanitization.
+	 * @param bool   $strict       Whether to limit the sanitization to specific characters. Default false.
+	 * @return string
+	 */
```

Comment:
This doc block should mention that all the logic is performed in WordPress' [remove_accents()](https://github.com/WordPress/wordpress-develop/blob/c7783dc49a016b98480d2966b14a59674bf41588/src/wp-includes/formatting.php#L1584) function.